src/main/fc/core.c

```diff
@@ -319,12 +316,22 @@ void updateArmingStatus(void)
             unsetArmingDisabled(ARMING_DISABLED_THROTTLE);
         }
 
-        if (!isUpright() && !IS_RC_MODE_ACTIVE(BOXFLIPOVERAFTERCRASH)) {
+        if (!isUpright() && !IS_RC_MODE_ACTIVE(BOXCRASHFLIP)) {
             setArmingDisabled(ARMING_DISABLED_ANGLE);
         } else {
             unsetArmingDisabled(ARMING_DISABLED_ANGLE);
         }
 
+        // handle disabling turtle switch while crashflip is active as a disarm, unless
+        // we know that the quad flipped successfully, in which case we can fly away
+        if (crashFlipModeActive && !IS_RC_MODE_ACTIVE(BOXCRASHFLIP) && !crashFlipSuccessful()) {
+            crashFlipModeActive = false;
+            // stop arming (require a disarm/rearm cycle)
+            setArmingDisabled(ARMING_DISABLED_CRASHFLIP);
+        } else {
+            unsetArmingDisabled(ARMING_DISABLED_CRASHFLIP);
```

Comment:
doesn't this unset ARMING_DISABLED_CRASHFLIP on next iteration?

Reply:
Yes, however it has achieved its purpose, once the mode has changed, I think.
The only need for these lines is to 'hold' the disarm for the relatively unusual situation where we need the quad to stay disarmed after the switch change.  If not for the arming disabled flag, every time `tryArm()` runs, it will always arm the quad.  We would put a 'return' in `tryArm()`, before the point when the arming happens, but that might introduce other problems.

Reply:
Well, it does work, ARMING_DISABLED_CRASHFLIP is set for cycle cycle, but ARMING_DISABLED_ARM_SWITCH does latch.
It would be slightly cleaner with special variable to accumulate these single use ARMING_DISABLED flags - there will be no need  for unsetArmingDisabled and it would be intermittently visible from outside.

Reply:
I agree.
The whole way the arming/disarming code is structured makes it incredibly difficult to follow.
I had been concerned that since the Arm switch must be active while Crashflip is active, then there would be an arming_disabled flag set by just being armed, and that we would not be able to use the method above to bypass tryArm() and retain the disarm that we wanted.  However, it worked. I don't quite understand how/why.